core/kazoo_services/src/kz_services_recurring.erl

```diff
@@ -0,0 +1,379 @@
+-spec send_reminder(kz_term:ne_binary(), non_neg_integer()) -> {'ok', kz_term:ne_binary()}.
+send_reminder(AccountId, DueTimestamp) ->
+    FetchOptions = ['hydrate_account_quantities'
+                   ,'hydrate_cascade_quantities'
+                   ,'hydrate_plans'
+                   ,'hydrate_invoices'
+                   ,'skip_cache'
+                   ],
+    Services = kz_services:fetch(AccountId, FetchOptions),
+    Invoices = kz_services:invoices(Services),
+    _ = kz_services_invoices:foldl(fun(Invoice, _Acc) ->
+                                           do_notify_reseller(AccountId, Services, Invoice, DueTimestamp)
+                                   end
+                                  ,'ok'
+                                  ,Invoices
+                                  ),
+    {'ok', status_good()}.
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% @end
+%%------------------------------------------------------------------------------
+-spec do_notify_reseller(kz_term:ne_binary(), kz_services:services(), kz_services_invoice:invoice(), non_neg_integer()) -> 'ok'.
+do_notify_reseller(AccountId, Services, Invoice, DueTimestamp) ->
+    lager:debug("sending bill reminder notification to account ~s", [AccountId]),
+    Props = [{<<"Account-ID">>, AccountId}
+            ,{<<"Due-Date">>, DueTimestamp}
+            ,{<<"Items">>
+             ,kz_services_items:public_json(
+                kz_services_invoice:items(Invoice)
+               )
+             }
+            ,{<<"Timestamp">>, kz_time:now_s()}
+             | maybe_add_payment_token(Services, Invoice)
+             ++ kz_api:default_headers(?APP_NAME, ?APP_VERSION)
+            ],
+    kapps_notify_publisher:cast(props:filter_undefined(Props), fun kapi_notifications:publish_bill_reminder/1).
+
+-spec maybe_add_payment_token(kz_services:services(), kz_services_invoice:invoice()) ->
+                                     kz_term:proplist().
+maybe_add_payment_token(Services, Invoice) ->
+    case kz_services_invoice:bookkeeper_type(Invoice) of
+        'undefined' -> [];
+        Bookkeeper ->
+            [{<<"Payment-Token">>, kz_services_payment_tokens:default(Services, Bookkeeper)}]
+    end.
+
+%%%=============================================================================
+%%% Internal functions
+%%%=============================================================================
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% @end
+%%------------------------------------------------------------------------------
+-spec get_previous_month(kz_time:gregorian_seconds()) -> kz_time:date().
+get_previous_month(DueTimestamp) ->
+    {{DueYear, DueMonth, _}, _} = calendar:gregorian_seconds_to_datetime(DueTimestamp),
+    calendar:datetime_to_gregorian_seconds({kz_date:normalize({DueYear, DueMonth - 1,1}), {0, 0, 1}}).
```

Comment:
`calendar` returns gregorian seconds but the type (and expected return where called is a date 3-tuple). How does this work?